tool/src/cpp2/ty.rs

```diff
@@ -280,11 +283,13 @@ impl<'ccx, 'tcx: 'ccx, 'header> TyGenContext<'ccx, 'tcx, 'header> {
         let c_header = self.c.gen_struct_def(def);
         let c_impl_header = self.c.gen_impl(def.into());
 
+        self.generating_struct_fields = true;
         let field_decls = def
             .fields
             .iter()
             .map(|field| self.gen_ty_decl(&field.ty, field.name.as_str()))
             .collect::<Vec<_>>();
+        self.generating_struct_fields = false;
```

Comment:
Thought: code might be cleaner if you had a `TyDeclOptions` struct that you passed down through `gen_ty_decl` instead of turning a flag on the `self` on and off

Reply:
Part of the point of TyGenContext is to avoid having to pass down a bunch of stuff.